src/backend/mod.rs

```diff
@@ -184,14 +186,78 @@ pub trait Backend: Sync + Send {
 
 impl Serialize for dyn Backend {
     fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        fn count_white_spaces(str: &str) -> i32 {
+            let mut white_spaces: i32 = 0;
+            for ch in str.to_string().chars() {
+                if ch.is_whitespace() {
+                    white_spaces += 1;
+                } else {
+                    break;
+                }
+            }
+            white_spaces
+        }
+
+        fn rec_insert(mut val: &mut Value, datum_json: Value, counter: i32) {
+            //each unwrap will never fail
+            if counter != 1 {
+                rec_insert(val.get_mut("data").unwrap(), datum_json, counter - 1)
+            } else {
+                let is_object = val.is_object();
+
+                if is_object {
+                    let last_prop = val.as_object_mut().unwrap();
+                    last_prop
+                        .get_mut("data")
+                        .unwrap()
+                        .as_array_mut()
+                        .unwrap()
+                        .push(datum_json);
+                } else {
+                    let last_prop = val.as_array_mut().unwrap();
+
+                    if last_prop.is_empty() {
+                        last_prop.push(datum_json);
+                    } else {
+                        val = last_prop.last_mut().unwrap();
+                        rec_insert(val, datum_json, counter)
+                    }
+                }
+            }
+        }
+
+        //Logic
```

Comment:
Logic what?

Reply:
Just for anyone trying to understand the code, that the overall logic can be seen from here.